Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -6,50 +6,640 @@
+            {
+                if (buttonId == m_transformTranslateButtonId)
+                {
+                    m_transformType = TransformType::Translation;
+                }
+                else if (buttonId == m_transformRotateButtonId)
+                {
+                    m_transformType = TransformType::Rotation;
+                }
+                else if (buttonId == m_transformScaleButtonId)
+                {
+                    m_transformType = TransformType::Scale;
+                }
+
+                RefreshManipulator();
+            });
+
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::RegisterClusterEventHandler,
+            m_transformClusterId,
+            m_transformSelectionHandler);
+
+        RefreshManipulator();
+    }
+
+    TransformMode::~TransformMode()
+    {
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::RemoveCluster,
+            m_transformClusterId);
+
+        DestroyManipulators();
+    }
+
+    void TransformMode::DestroyManipulators()
+    {
+        if (m_manipulator)
+        {
+            m_manipulator->Unregister();
+            m_manipulator.reset();
+        }
+    }
+
     void TransformMode::Refresh()
     {
+        m_whiteBoxSelection.reset();
+        DestroyManipulators();
     }
 
-    AZStd::vector<AzToolsFramework::ActionOverride> TransformMode::PopulateActions([[maybe_unused]] const AZ::EntityComponentIdPair& entityComponentIdPair)
+    AZStd::vector<AzToolsFramework::ActionOverride> TransformMode::PopulateActions(
+        [[maybe_unused]] const AZ::EntityComponentIdPair& entityComponentIdPair)
     {
-        return {};
+        return {
+            AzToolsFramework::ActionOverride()
+                .SetUri(SwitchTranslationMode)
+                .SetKeySequence(QKeySequence{Qt::Key_1})
+                .SetTitle(SwitchToTranslationModeTile)
+                .SetTip(SwitchToTranslationModeDesc)
+                .SetEntityComponentIdPair(entityComponentIdPair)
+                .SetCallback(
+                    [cluserId = m_transformClusterId, buttonId = m_transformTranslateButtonId]()
+                    {
+                        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+                            AzToolsFramework::ViewportUi::DefaultViewportId,
+                            static_cast<void(AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::*)(
+                                    AzToolsFramework::ViewportUi::ClusterId, 
+                                    AzToolsFramework::ViewportUi::ButtonId)>(
+                                &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::PressButton),
```

Comment:
@hultonha you need the static cast since there are multiple implementations of PressButton.

```
        //! Invoke a button press on a cluster.
        virtual void PressButton(ClusterId clusterId, ButtonId buttonId) = 0;
        //! Invoke a button press on a switcher.
        virtual void PressButton(SwitcherId switcherId, ButtonId buttonId) = 0;
```

Reply:
Aaah interesting 🤔 There actually might be a nicer way to do this using a lambda, I know @lumberyard-employee-dm often recommends this, might make the calling syntax a bit cleaner

Reply:
Yes that can be done by supplying a lambda whose first parameters is of the type of the EBus Interface pointer
```suggestion
                            [](AzToolsFramework::ViewportUi::ViewportUiRequests* requests)
                            {
                                requests->PressButton();
                            };
```

Reply:
That was it! 😅 Thanks for adding the suggestion @lumberyard-employee-dm 👍

Reply:
umm kind of questionable if its that much shorter